controllers/replica_set_controller.go

```diff
@@ -442,17 +442,20 @@ func (r *ReplicaSetReconciler) deployMongoDBReplicaSet(mdb mdbv1.MongoDBCommunit
 // The Service definition is built from the `mdb` resource. If `isArbiter` is set to true, the Service
 // will be created for the arbiters Statefulset.
 func (r *ReplicaSetReconciler) ensureService(mdb mdbv1.MongoDBCommunity) error {
-	name := mdb.ServiceName()
+	processPortManager, err := r.createProcessPortManager(mdb)
+	if err != nil {
+		return err
+	}
 
-	svc := &corev1.Service{ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: mdb.Namespace}}
+	svc := &corev1.Service{ObjectMeta: metav1.ObjectMeta{Name: mdb.ServiceName(), Namespace: mdb.Namespace}}
 	op, err := controllerutil.CreateOrUpdate(context.TODO(), r.client, svc, func() error {
 		resourceVersion := svc.ResourceVersion // Save resourceVersion for later
-		*svc = buildService(mdb)
+		*svc = r.buildService(mdb, processPortManager)
 		svc.ResourceVersion = resourceVersion
```

Comment:
I think because service builder does not populate this field and in docs there is:
```
	// Clients must treat these values as opaque and passed unmodified back to the server.
```
https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L171
But I could be mistaken.

Reply:
Isn't this field changed automatically by the api server when anything was changed in values?

Reply:
I tried to play with `kubectl create` and `kubectl replace` (the `kubectl apply` is a bad choice as it records its own annotation with the last seen configuration) and it seems those values are always replaced by the API Server. I would need to dig deeper into it but at a first glance, calling `svc.ResourceVersion = resourceVersion` has no effect.

Reply:
@slaskawi Would you like to investigate it further? For now in this PR I'll leave it as is.

Reply:
No, it's fine to leave it as is in this PR. If we ever wanted to change this, it should have it's own PR.